src/main/java/org/openlmis/stockmanagement/web/OrganizationController.java

```diff
@@ -106,7 +106,8 @@ private void checkIsValidUpdateModel(UUID id, Organization organization) {
 
   private void checkUpdateOrganizationDuplicate(Organization organization) {
     Organization foundByName = organizationRepository.findByName(organization.getName());
-    if (foundByName != null && foundByName.getId() != organization.getId()) {
+    if (foundByName != null && foundByName.getId() != organization.getId()
```

Comment:
There was a bug that did not allow updating the entity (== instead of equals() for UUID comparison). I fixed the error.

Reply:
Where was that bug, please provide location.

Reply:
Here:
https://github.com/OpenLMIS/openlmis-stockmanagement/pull/47/files#diff-ee3912bb6f357d88aefaee14ef999afbcdaccc27cf41986a96da660b200c88e5R109

Reply:
Good job with that check! However, we should assume that organization from update request will never have id set as the id goes through path. So id has to be set to this organization before calling this method.

Reply:
I agree. Currently, the algorithm allows you to update an entity to give it the same name as another existing entity, which later leads to errors such as:

> "query did not return a unique result: 2; nested exception is javax.persistence.NonUniqueResultException: query did not return a unique result: 2"

I moved the line
`organization.setId(id);`
above the line
`checkIsValidUpdateModel(id, organization);`
This will ensure that the entity with the ID provided in the endpoint's path will be updated and will avoid name duplication.

Reply:
Very well!